distributed/tests/test_scheduler.py

```diff
@@ -4670,14 +4673,69 @@ def block_on_event(input, block, executing):
     assert s.queued
     await s.remove_worker(address=a.address, stimulus_id="test")
 
-    s.validate_state()
+    if validate:
+        s.validate_state()
 
     await block.set()
     await executing.clear()
 
     async with Worker(s.address) as b:
+        if validate:
+            s.validate_state()
+        await c.gather(futs)
+        if validate:
+            s.validate_state()
+
+
+@pytest.mark.skipif(
+    not QUEUING_ON_BY_DEFAULT,
+    reason="The situation handled in this test requires queueing.",
+)
+@pytest.mark.parametrize("validate", [True, False])
+@gen_cluster(client=True)
+async def test_deadlock_dependency_of_queued_released_when_worker_removed(
+    c, s, a, b, validate
+):
+    @delayed
+    def inc(input):
+        return input + 1
+
+    @delayed
+    def block_on_event(input, block, executing):
+        executing.set()
```

Comment:
you aren't actually using executing anywhere

Reply:
Good catch, I dropped that in the process of adjusting this test.